static/app/views/settings/account/accountAuthorizations.tsx

```diff
@@ -24,98 +27,92 @@ type Authorization = {
   scopes: string[];
 };
 
-type Props = RouteComponentProps<{}, {}>;
+function AccountAuthorizations() {
+  const api = useApi();
+  const queryClient = useQueryClient();
+  const ENDPOINT = '/api-authorizations/';
 
-type State = {
-  data: Authorization[];
-} & DeprecatedAsyncView['state'];
+  const {data, isLoading, isError, refetch} = useApiQuery<Authorization[]>([ENDPOINT], {
+    staleTime: 0,
+  });
 
-class AccountAuthorizations extends DeprecatedAsyncView<Props, State> {
-  getEndpoints(): ReturnType<DeprecatedAsyncView['getEndpoints']> {
-    return [['data', '/api-authorizations/']];
+  if (isLoading) {
+    return <LoadingIndicator />;
   }
 
-  getTitle() {
-    return 'Approved Applications';
+  if (isError) {
+    return <LoadingError onRetry={refetch} />;
   }
 
-  handleRevoke = authorization => {
-    const oldData = this.state.data;
-
-    this.setState(
-      state => ({
-        data: state.data.filter(({id}) => id !== authorization.id),
-      }),
-      async () => {
-        try {
-          await this.api.requestPromise('/api-authorizations/', {
-            method: 'DELETE',
-            data: {authorization: authorization.id},
-          });
-
-          addSuccessMessage(t('Saved changes'));
-        } catch (_err) {
-          this.setState({
-            data: oldData,
-          });
-          addErrorMessage(t('Unable to save changes, please try again'));
-        }
-      }
+  const handleRevoke = (authorization: Authorization) => {
+    const oldData = data;
+    setApiQueryData<any>(queryClient, [ENDPOINT], prevData =>
+      prevData.filter(a => a.id !== authorization.id)
     );
+
+    async () => {
+      try {
+        await api.requestPromise('/api-authorizations/', {
+          method: 'DELETE',
+          data: {authorization: authorization.id},
+        });
+        addSuccessMessage(t('Saved changes'));
+      } catch (_err) {
+        setApiQueryData<any>(queryClient, [ENDPOINT], oldData);
```

Comment:
wasn't sure if i could do this or if i need something like

`setApiQueryData<any>(queryClient, [ENDPOINT], _ => oldData);`